### PR TITLE
x86_semantics: Refactor sexpr output to use sexpr package

### PR DESCRIPTION
**Pull request by dagit**

### CODE REVIEW

File: lean/deps/x86_semantics/src/x86_semantics/common.lean [hunk 1]
```diff
@@ -1,26 +1,112 @@
...
+/-- Symbol characters allowed in simple symbols -/
+protected
+def char_symbols : list char :=
+ ['~', '!', '@', '$', '%', '^', '&', '*', '_', '-', '+', '=', '<', '>', '.', '?', '/' ]
+
+/-- Predicate that checks if character is allowed in a simple symbol. -/
+inductive is_symbol_char (c:char) : Prop
+| is_alpha : char.is_alpha c → is_symbol_char
+| is_digit : char.is_digit c → is_symbol_char
+| is_other : c ∈ sexp.char_symbols → is_symbol_char
```
Comment (joehendrix) [inline lean/deps/x86_semantics/src/x86_semantics/common.lean]: This looks copy pasted from SMTLIB.  Do we need all these special symbols in our x86 semantics?

Comment (dagit) [reply lean/deps/x86_semantics/src/x86_semantics/common.lean]: It was and I don't think so. Currently the only non-alphanumeric that we use is `$`.

How do you want me to proceed?

Comment (joehendrix) [reply lean/deps/x86_semantics/src/x86_semantics/common.lean]: Maybe just change the list to include '$'.

Comment (joehendrix) [reply lean/deps/x86_semantics/src/x86_semantics/common.lean]: You may want underscore as well.

Comment (dagit) [reply lean/deps/x86_semantics/src/x86_semantics/common.lean]: Done.

**Pull request merged**